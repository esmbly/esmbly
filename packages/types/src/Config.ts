import { Output } from './Output';

export interface Config {
  input: string[];
  transformers: string | Transformer;
  output: string | Output;
}
