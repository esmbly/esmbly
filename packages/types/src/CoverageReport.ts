export interface CoverageRange {
  startOffset: number;
  endOffset: number;
}

export interface FunctionCoverage {
  functionName: string;
  ranges: CoverageRange[];
  isBlockCoverage: boolean;
}

export interface CoverageReport {
  scriptId: string;
  url: string;
  functions: FunctionCoverage[];
}
