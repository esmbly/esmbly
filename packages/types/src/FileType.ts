export enum FileType {
  JavaScript = '.js',
  TypeScript = '.ts',
  WebAssembly = '.wasm',
}
