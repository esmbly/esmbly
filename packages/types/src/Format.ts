export enum Format {
  Flow = 'Flow',
  TypeScript = 'TypeScript',
  WebAssembly = 'WebAssembly',
  Wat = 'Wat',
  Asm = 'Asm',
  JSDoc = 'JSDoc',
  Any = 'Any',
  AssemblyScript = 'AssemblyScript',
  TSDefinition = 'TSDefinition'
}
