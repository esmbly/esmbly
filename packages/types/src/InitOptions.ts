export interface InitOptions {
  default: boolean;
  force: boolean;
}
