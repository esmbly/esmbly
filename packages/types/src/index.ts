export * from './AST';
export * from './File';
export * from './FileType';
export * from './RunConfig';
export * from './Transformer';
export * from './Config';
export * from './Output';
export * from './InitOptions';
