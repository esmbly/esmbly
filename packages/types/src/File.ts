import { FileType } from './FileType';
import { Output } from './Output';

export interface File {
  name: string;
  content: string | Buffer;
  dir: string;
  type: FileType;
  outputOptions?: Output
}
