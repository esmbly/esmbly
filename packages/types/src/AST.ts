import * as recast from 'recast';
import { File } from './File';
import { FileType } from './FileType';

export interface AST {
  name: string;
  path: string;
  tree: recast.ASTNode;
  type: FileType;
  toFile(): File;
}
