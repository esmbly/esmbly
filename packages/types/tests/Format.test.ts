import { Format } from '../src';

describe('Format', () => {
  it('exposes the correct enum', () => {
    expect(Format).toMatchSnapshot();
  });
});
