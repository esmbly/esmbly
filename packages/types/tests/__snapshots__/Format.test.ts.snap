// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format exposes the correct enum 1`] = `
Object {
  "Any": "Any",
  "Asm": "Asm",
  "AssemblyScript": "AssemblyScript",
  "Flow": "Flow",
  "JSDoc": "JSDoc",
  "TypeScript": "TypeScript",
  "Wat": "Wat",
  "WebAssembly": "WebAssembly",
}
`;
